src/PKSim.Core/Services/ParameterContainerTask.cs

```diff
@@ -51,30 +56,38 @@ public ParameterContainerTask(IParameterQuery parameterQuery, IParameterFactory
 
       public void AddIndividualParametersTo<TContainer>(TContainer parameterContainer, OriginData originData, string parameterName) where TContainer : IContainer
       {
-         addParametersTo(parameterContainer, originData, originData.AllCalculationMethods().Select(cm => cm.Name), 
+         addParametersTo(parameterContainer, originData, originData.AllCalculationMethods().Select(cm => cm.Name),
             param => param.BuildingBlockType == PKSimBuildingBlockType.Individual && string.Equals(param.ParameterName, parameterName));
       }
 
-      public void AddDiseaseStateParametersTo(DiseaseState diseaseState)
-      {
-         addParametersTo(diseaseState, null, CoreConstants.CalculationMethod.ForDiseaseStates);
-      }
+      public void AddDiseaseStateParametersTo(DiseaseState diseaseState) => addParametersTo(diseaseState, calculationMethods: CoreConstants.CalculationMethod.ForDiseaseStates);
 
-      public void AddModelParametersTo<TContainer>(TContainer parameterContainer, OriginData originData, ModelProperties modelProperties, IFormulaCache formulaCache) where TContainer : IContainer
+      public void AddParametersToSpatialStructureContainer<TContainer>(TContainer parameterContainer, OriginData originData, ModelProperties modelProperties, IFormulaCache formulaCache) where TContainer : IContainer
       {
-         bool addParameter(ParameterMetaData param)
+         bool addParameter(ParameterMetaData parameterMetaData)
          {
-            return param.BuildingBlockType != PKSimBuildingBlockType.Individual || _commonParametersRepository.UsedForAllSpecies(param);
+            //all non individual parameters are added to the spatial structure by default
+            //otherwise, we add all parameters that are used by all species. The value used will be set based on the fact that the parameter has a shared value for all parameters or species or not
+            return parameterMetaData.BuildingBlockType != PKSimBuildingBlockType.Individual || _individualParameterBySpeciesRepository.UsedForAllSpecies(parameterMetaData);
```

Comment:
@Yuri05 @rwmcintosh 
I think this is it. We add a parmaeter to the spt structure if it's either not an individual parameter OR if it's used by all species? This calls returns true for distributed parameters

Maybe this is right that it returns true. But this makes thinks very complicated then because we have different parameter structure afterwards (distributed vs non distrebitued)

Reply:
Is solved now (distributed parameters not returned as common for all species anymore)